applications/nrf_desktop/doc/watchdog.rst

```diff
@@ -0,0 +1,43 @@
+.. _watchdog:
+
+Watchdog module
+###############
+
+The ``watchdog`` module is responsible for controlling the watchdog timer.
+The watchdog timer is used to prevent hangs caused by software or hardware faults.
+When the system hangs, it is automatically restarted by the watchdog timer.
+
+Module Events
+*************
+
++----------------+-------------+--------------+-----------------+------------------+
+| Source Module  | Input Event | This Module  | Output Event    | Sink Module      |
++================+=============+==============+=================+==================+
+|                |             | ``watchdog`` |                 |                  |
++----------------+-------------+--------------+-----------------+------------------+
+
+External dependencies
+*********************
+
+The module uses the Zephyr :ref:`zephyr:watchdog` driver.
+For this reason you should set :option:`CONFIG_WATCHDOG` option.
+
+Configuration
+*************
+
+The module is enabled by the ``CONFIG_DESKTOP_WATCHDOG_ENABLE`` option.
+
+You must define ``CONFIG_DESKTOP_WATCHDOG_TIMEOUT``.
+After this amount of time (in ms), the device will be restarted if the watchdog timer was not fed.
+
+.. note::
+  The module is used only in the release configurations (``ZRelease``, ``ZReleaseMCUBoot``).
+  For the debug configurations (``ZDebug``, ``ZDebugWithShell``, ``ZDebugMCUBoot``) enabling watchdog timer may cause losing logs, for example when logger is in panic mode.
+
+Implementation details
+**********************
+
+The watchdog timer is started on the system start.
+The module periodically feeds the watchdog timer using :c:type:`struct k_delayed_work`.
+The work resubmits itself with delay equal to ``CONFIG_DESKTOP_WATCHDOG_TIMEOUT / 3``.
+In case of the system hang, the work will not be processed, the watchdog timer will not be fed in time and the system will be restarted.
```

Comment:
```suggestion
In case of the system hang, the work will not be processed, the watchdog timer will not be fed with data on time, and the system will be restarted.
```

Reply:
Feeding the watchdog (also refered as kicking the watchdog) is resetting the watchdog timer. The watchdog is not fed with data.

Reply:
In the documentation I will use `reset the watchdog timer` instead of `feed the watchdog timer`